File: web/components/tabs-section.dart
/// <reference path="../../tsd.d.ts" />
import "package:angular2/angular2.dart"
    show Component, View, CORE_DIRECTIVES, NgNonBindable;
import "../../lib/index.dart" show tabs;
import "tabs/tabs-demo.dart" show TabsDemo;

var name = "Tabs";

var src = "https://github.com/valor-software/ng2-bootstrap/blob/master/components/tabs/tabs.ts";
// webpack html imports
var doc = require("../../lib/tabs/readme.md");

var titleDoc = require("../../lib/tabs/title.md");

var ts = require("!!prismjs?lang=typescript!./tabs/tabs-demo.ts");

var html = require("!!prismjs?lang=markup!./tabs/tabs-demo.html");

@Component (selector: "tabs-section")
@View (template: '''
  <br>
  <section id="${ name.toLowerCase()}">
    <div class="row"><h1>${ name}<small>(<a href="${ src}">src</a>)</small></h1></div>

    <hr>

    <div class="row"><div class="col-md-12">${ titleDoc}</div></div>

    <div class="row">
      <h2>Example</h2>
      <div class="card card-block panel panel-default panel-body">
        <tabs-demo></tabs-demo>
      </div>
    </div>

    <br>

    <div class="row">
      <tabset>
        <tab heading="Markup">
          <div class="card card-block panel panel-default panel-body">
            <pre class="language-html"><code class="language-html" ng-non-bindable>${ html}</code></pre>
          </div>
        </tab>
        <tab heading="TypeScript">
          <div class="card card-block panel panel-default panel-body">
            <pre class="language-typescript"><code class="language-typescript" ng-non-bindable>${ ts}</code></pre>
          </div>
        </tab>
      </tabset>
    </div>

    <br>

    <div class="row">
      <h2>API</h2>
      <div class="card card-block panel panel-default panel-body">${ doc}</div>
    </div>
  </section>
  ''', directives: const [ TabsDemo, tabs, CORE_DIRECTIVES, NgNonBindable])
class TabsSection {}

File: demo/components/tabs-section.ts
import {Component} from '@angular/core';
import {CORE_DIRECTIVES} from '@angular/common';

import {TabsDemoComponent} from './tabs/tabs-demo';
import {DemoSectionComponent} from './demo-section';

// webpack html imports
let doc = require('../../components/tabs/readme.md');
let titleDoc = require('../../components/tabs/title.md');

let ts = require('!!prismjs?lang=typescript!./tabs/tabs-demo.ts');
let html = require('!!prismjs?lang=markup!./tabs/tabs-demo.html');

@Component({
  selector: 'tabs-section',
  directives: [DemoSectionComponent, TabsDemoComponent, CORE_DIRECTIVES],
  template: `
    <demo-section [name]="name" [src]="src" [titleDoc]="titleDoc" [html]="html" [ts]="ts" [doc]="doc">
      <tabs-demo></tabs-demo>
    </demo-section>`
})
export class TabsSectionComponent {
  public name:string = 'Tabs';
  public src:string = 'https://github.com/valor-software/ng2-bootstrap/blob/master/components/tabs';
  public html:string = html;
  public ts:string = ts;
  public titleDoc:string = titleDoc;
  public doc:string = doc;
}
